### PR TITLE
Fix: mapping datasets bounds & tooltips

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

- Using total_bounds instead of union_all because it's simpler and union can fail on some geometries.
- Added `TooltipOptions` to provide additional formatting information for the map tooltips.

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases) - No
- [x] Has documentation (including user guide) been updated - No
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt` - NA
- [x] Code linting, tests and other workflows pass

### CODE REVIEW

File: src/caf/viz/web/mapping.py [hunk 1]
```diff
@@ -338,7 +371,7 @@ def map_datasets(
     folium.LayerControl(collapsed=False).add_to(map_)
 
     # Fit map to bounds of mask or last dataset
-    bounds = Bounds(*mask.bounds) if mask else Bounds(*data.union_all().bounds)
+    bounds = Bounds(*mask.bounds) if mask else Bounds(*data.total_bounds)
```
Comment (amber-mertens-tfn) [inline src/caf/viz/web/mapping.py:374]: Think I did that because I struggled to get a good result that opened the map zoomed into the data. But this still seems to zoom correctly in the example anyway, so all good.

**Review by amber-mertens-tfn — APPROVED** [review]
